runners/core-construction-java/src/test/java/org/apache/beam/runners/core/construction/graph/GreedyPipelineFuserTest.java

```diff
@@ -1144,4 +1144,135 @@ public void compositesIgnored() {
                 .withNoOutputs()
                 .withTransforms("goTransform")));
   }
+
+  @Test
+  public void sanitizedTransforms() throws Exception {
+
+    PCollection flattenOutput = pc("flatten.out");
```

Comment:
I was going to suggest to build the pipeline using the pattern that we are hitting right now where Flatten has dangling inputs instead of creating the PTransforms manually.

```
Pipeline p = Pipeline.create();
PCollection<String> pc1 = ...
PCollection<String> pc2 = ...
PCollection<String> flatten = Flatten...
...
```

This will exercise a real pipeline instead of having the fake.

Reply:
I added a separate test on the real pipeline https://github.com/apache/beam/pull/6132/files/fb15e313835b5b4a4b60fbc1c82a24112a7b1cf0#diff-0f688c9cfedd2fbf479c61aa26f4dc17R716
I was trying to follow the similar pattern of testing as done for the other tests in this file.
Let me know if I should modify the test. I can take it in a separate PR.

Reply:
Sorry, missed that for some reason. Anyways this LGTM as is.